### ISSUE TITLE
Pocketing entities

### ISSUE THREAD

**Issue by mcNuggets1** [FPtje/DarkRP#2774]
[//]: # (DO NOT REMOVE THIS TEMPLATE.)
[//]: # (GITHUB ISSUES ARE **NOT** FOR HELP)
[//]: # (GO TO THE FORUMS INSTEAD: http://forum.darkrp.com/ )

### Description of the bug
[//]: # (Describe the issue as accurately as possible)
When pocketing entities, you can bypass the max entities limit from the F4 menu.

### How to make the bug happen
[//]: # (Knowing how to make a bug happen allows the developers to figure out)
[//]: # (what the cause of the problem is and whether a certain fix solves it)
Pocket a entity which has a max setting on it, spawn it again in the F4 menu.

### Lua errors
[//]: # (Note that errors on server startup are more important than other ones)
[//]: # (because they can cause other errors. Always look in the startup log of the server!)
none

### Why the developer of DarkRP is responsible for this issue
[//]: # (By posting on Github, you ask the developers of DarkRP to solve the problem)
[//]: # (It may seem obvious to you, but you have to make clear why they are the right people to look at it)
his gamemode

I would suggest we use a table and add all entities which has the Getowning_ent var entity on it and check the table when trying to buy entities, this would fix the mechanic accurately.

**Comment by FPtje** [comment]
This wasn't possible at some point.

**Comment by mcNuggets1** [comment]
There is always a solution. Especially for this issue, there should be plenty.

**Comment by mcNuggets1** [comment]
I'll take a look at it.

**Comment by FPtje** [comment]
No I mean there should be some solution, it might have broken at some point, but there should be some code there. I remember fixing this

**Comment by mcNuggets1** [comment]
Ohhh. Well, it's been around for atleast 4 years now.

**Comment by mcNuggets1** [comment]
As far as I can see, addCustomEntity (or whatever it is called) adds the entity to the list. Making the count work.

But because of EntityRemoved calls removeCustomEntity and the entity is getting removed when pocketed, the count lowers.

**Comment by mcNuggets1** [comment]
A quick fix could be to add a var to the entity before being removed when pocketed which prevents calling removeCustomEntity.

**Comment by Bo98** [comment]
Pickup from the pocket does an extra `addCustomEntity` call which should counter the `removeCustomEntity` call: https://github.com/FPtje/DarkRP/blob/master/entities/weapons/pocket/sv_init.lua#L277

Related: #1508

**Comment by mcNuggets1** [comment]
It only works when owning_ent is written to it.
My fix would work even for entities not having it.
Maybe fixing the bug altogether

**Comment by Bo98** [comment]
Ah, so it only affects certain entities. It was just missing the check for `ent.SID` then, which was added to the remove logic at a later date.

**Comment by matiasg90** [comment]
this update complicated the jobs as doctors who had an entity limit
now they can only buy 2 medical kits and until the users do not use it, they can not buy again

**Comment by FPtje** [comment]
You mean they previously circumvented the limit by putting them in their pockets. That's not the right way of going at it. What behaviour do you want with these med kits? Why do you set a limit of 2 in the first place?

**Comment by matiasg90** [comment]
You're right, a long time ago I use limit 2 medical kit. I will increase it to 20 to prevent them from bothering with the collision of entities
